hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -289,33 +289,60 @@ private void processRecords(ManagedRocksIterator iterator,
     long count = 0;
     List<Future<Void>> futures = new ArrayList<>();
     boolean reachedEnd = false;
+
+    Map<String, Filter> fieldsFilterSplitMap = new HashMap<>();
+    if (filter != null) {
+      for (String field : filter.split(",")) {
```

Comment:
what is being split by ``,`` here, your filter would be like ``dataSize:lesser:500`` and you can't specify multiple filters by this code if you do so something like ``dataSize:lesser:500, dataSize:equal:500`` this crashes.

Can you help what is this doing? & maybe extend a test case which uses this, if not already

I removed this for loop itself & no test fails in ``TestLDBCLI``, I guess the intention might be to support multiple filters maybe, if so you need to fix it

Reply:
The aim of having this loop is to pass multiple filters like `dataSize:lesser:500, bucketName:equal:bucket1`. It is to apply filters to different fields, not on the same field.
I will add some more tests to cover the scenario.